catalog/plugin/catalog-plugin-duplication/src/main/java/ddf/catalog/metacard/duplication/DuplicationValidator.java

```diff
@@ -0,0 +1,303 @@
+        this.catalogFramework = catalogFramework;
+        this.filterBuilder = filterBuilder;
+    }
+
+    /**
+     * Setter for the list of attributes to test for duplication in the local catalog.  Resulting
+     * attributes will cause the {@link ddf.catalog.data.impl.BasicTypes#VALIDATION_ERRORS} attribute
+     * to be set on the metacard.
+     *
+     * @param attributeStrings
+     */
+    public void setErrorOnDuplicateAttributes(String[] attributeStrings) {
+        if (attributeStrings != null) {
+            this.errorOnDuplicateAttributes = Arrays.copyOf(attributeStrings,
+                    attributeStrings.length);
+        }
+    }
+
+    /**
+     * Setter for the list of attributes to test for duplication in the local catalog.  Resulting
+     * attributes will cause the {@link ddf.catalog.data.impl.BasicTypes#VALIDATION_WARNINGS} attribute
+     * to be set on the metacard.
+     *
+     * @param attributeStrings
+     */
+    public void setWarnOnDuplicateAttributes(String[] attributeStrings) {
+        if (attributeStrings != null) {
+            this.warnOnDuplicateAttributes = Arrays.copyOf(attributeStrings,
+                    attributeStrings.length);
+        }
+    }
+
+    @Override
+    public Optional<MetacardValidationReport> validateMetacard(Metacard metacard) {
+        Preconditions.checkArgument(metacard != null, "The metacard cannot be null.");
+
+        return getReport(reportDuplicates(metacard));
+
+    }
+
+    @Override
+    public void validate(Metacard metacard) throws ValidationException {
+
+        final Optional<MetacardValidationReport> report = validateMetacard(metacard);
+
+        if (report.isPresent()) {
+            final List<String> errors = report.get()
+                    .getMetacardValidationViolations()
+                    .stream()
+                    .filter(validationViolation -> validationViolation.getSeverity()
+                            .equals(ValidationViolation.Severity.ERROR))
+                    .map(ValidationViolation::getMessage)
+                    .collect(Collectors.toList());
+            final List<String> warnings = report.get()
+                    .getMetacardValidationViolations()
+                    .stream()
+                    .filter(validationViolation -> validationViolation.getSeverity()
+                            .equals(ValidationViolation.Severity.WARNING))
+                    .map(ValidationViolation::getMessage)
+                    .collect(Collectors.toList());
+
+            String message = String.format("Duplicate data found in catalog for ID {%s}.",
+                    metacard.getId());
+            final ValidationExceptionImpl exception = new ValidationExceptionImpl(message);
+            exception.setErrors(errors);
+            exception.setWarnings(warnings);
+            throw exception;
+        }
+    }
+
+    private Set<ValidationViolation> reportDuplicates(final Metacard metacard) {
```

Comment:
Would you ever see overriding this method? e.g. we want to test for duplicates in internal records like workspaces, registry entries, etc. that may have different fields from resources
Maybe make these methods protected instead of private?